src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,986 @@
+                    return false;
+                ledgers_.push_back(ledger);
+            }
+
+            return true;
+        }
+    };
+
+    void
+    testLedgerData(testData& D, std::shared_ptr<Ledger> l, std::uint32_t seq)
+    {
+        using namespace test::jtx;
+
+        auto rootCount{0};
+        auto accCount{0};
+        auto sothCount{0};
+        for (auto const& s : l->sles)
+        {
+            if (s->getType() == ltACCOUNT_ROOT)
+            {
+                int sq = s->getFieldU32(sfSequence);
+                int reqsq = -1;
+                const auto id = s->getAccountID(sfAccount);
+                int i;
+                for (i = 0; i < D.A_.size(); ++i)
+                {
+                    if (id == D.A_[i]->id())
+                    {
+                        reqsq = ledgersPerShard + 1;
+                        for (int j = 0; j <= seq; ++j)
+                            if (D.acc_[j] > i + 1 ||
+                                (D.acc_[j] == i + 1 && !D.newAcc(j)))
+                            {
+                                for (int k = 0; k < D.pay_[j].size(); ++k)
+                                    if (D.pay_[j][k].first == i)
+                                        reqsq++;
+                            }
+                            else
+                                reqsq++;
+                        ++accCount;
+                        break;
+                    }
+                }
+                if (reqsq == -1)
+                {
+                    reqsq = D.acc_[seq] + 1;
+                    ++rootCount;
+                }
+                BEAST_EXPECT(sq == reqsq);
+            }
+            else
+                ++sothCount;
+        }
+        BEAST_EXPECT(rootCount == 1);
+        BEAST_EXPECT(accCount == D.acc_[seq]);
+        BEAST_EXPECT(sothCount == 3);
+
+        auto iniCount{0};
+        auto setCount{0};
+        auto payCount{0};
+        auto tothCount{0};
+        for (auto const& t : l->txs)
+        {
+            if (t.first->getTxnType() == ttPAYMENT)
+            {
+                std::int64_t xrp =
+                    t.first->getFieldAmount(sfAmount).xrp().decimalXRP();
+                if (xrp == iniAmount)
+                    ++iniCount;
+                else
+                {
+                    ++payCount;
+                    BEAST_EXPECT(xrp == D.xrp_[seq]);
+                }
+            }
+            else if (t.first->getTxnType() == ttACCOUNT_SET)
+                ++setCount;
+            else
+                ++tothCount;
+        }
+        int newacc = D.newAcc(seq) ? 1 : 0;
+        BEAST_EXPECT(iniCount == newacc);
+        BEAST_EXPECT(setCount == newacc);
+        BEAST_EXPECT(payCount == D.pay_[seq].size());
+        BEAST_EXPECT(tothCount == !seq);
+    }
+
+    bool
+    saveLedger(
+        Database* db,
+        std::shared_ptr<const Ledger> l,
+        std::shared_ptr<const Ledger> next = {})
```

Comment:
Couple nits:
1) `db` by reference
2) pass the two  `shared_ptr` by `const&` (passing by value requires incrementing atomic variables)
3) Do `l` and `next` need to be `shared_ptr`? Could be just pass them as `Ledger const&`?
4) `const` goes on the right. I.e. `Ledger const` not `const Ledger`

Reply:
Done